### PR TITLE
Bug fix for TEF exclusion not honored

### PR DESCRIPTION
**Pull request by vibhorgarg**

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -107,9 +107,9 @@
     </build>
 
     <properties>
-        <revision>0.1.5</revision>
+        <revision>0.2.0</revision>
```
Comment (bageshwar) [inline pom.xml:110]: This should be a minor version bump, since its a bugfix. Refer to semver for guidance.

Comment (vibhorgarg) [reply pom.xml:110]: This is also JDK17 compliant change. 
Hence the bump to major version seemed relevant.

File: tef-impl/src/test/java/flipkart/tef/execution/FluentCapabilityBuilderTest.java [hunk 2]
```diff
@@ -163,6 +164,62 @@ public List<Class<? extends IBizlogic>> exclusions() {
...
+                return list;
+            }
+        };
+
+        FluentCapabilityBuilder manager = new FluentCapabilityBuilder();
+        SimpleFlow flow = manager.withCapability(capabilityDefinition).dataflow();
+
+        assertNotNull(flow);
+
+        assertEquals(4, flow.getBizlogics().size());
```
Comment (bageshwar) [inline tef-impl/src/test/java/flipkart/tef/execution/FluentCapabilityBuilderTest.java:215]: Can you also assert on the order of bizlogics?

File: tef-impl/src/test/java/flipkart/tef/execution/FluentCapabilityBuilderTest.java [hunk 3]
```diff
@@ -163,6 +164,62 @@ public List<Class<? extends IBizlogic>> exclusions() {
...
+            }
+
+            @Override
+            public List<Class<? extends DataAdapterBizlogic>> adapters() {
+                return ImmutableList.of(DataAdapterBizlogic1.class);
+            }
+
+            @Override
+            public List<Class<? extends IBizlogic>> exclusions() {
+                return ImmutableList.of(DataAdapterBizlogic2.class);
```
Comment (bageshwar) [inline tef-impl/src/test/java/flipkart/tef/execution/FluentCapabilityBuilderTest.java:198]: Given adapter2 was never added to the flow from any place, this exclusion does not test things properly. I think you should add `DataAdapterBizlogic2` in `adapters`

**Review by bageshwar — CHANGES_REQUESTED** [review]

File: tef-impl/src/main/java/flipkart/tef/execution/FlowBuilder.java [hunk 4]
```diff
@@ -293,13 +296,13 @@ private void populateDataAdapterMap(Class<? extends IBizlogic> bizlogic) {
...
-            ParameterizedTypeImpl genericSuperClass = (ParameterizedTypeImpl) dataAdapterBizLogic.getGenericSuperclass();
+        if (dataAdapterBizLogic.getGenericSuperclass() instanceof ParameterizedType) {
+            ParameterizedType genericSuperClass = (ParameterizedType) dataAdapterBizLogic.getGenericSuperclass();
             if (genericSuperClass.getActualTypeArguments()[0] instanceof Class) {
                 return (Class<?>) genericSuperClass.getActualTypeArguments()[0];
-            } else if (genericSuperClass.getActualTypeArguments()[0] instanceof ParameterizedTypeImpl) {
+            } else if (genericSuperClass.getActualTypeArguments()[0] instanceof ParameterizedType) {
                 // The type itself is parameterized
-                return ((ParameterizedTypeImpl) genericSuperClass.getActualTypeArguments()[0]).getRawType();
+                return (Class<?>)((ParameterizedType) genericSuperClass.getActualTypeArguments()[0]).getRawType();
```
[the remark this answers is not in the document]
Comment (vibhorgarg) [reply tef-impl/src/main/java/flipkart/tef/execution/FlowBuilder.java:305]: getRawTypes, returns object implementing the Type interface.

File: tef-impl/src/test/java/flipkart/tef/execution/FluentCapabilityBuilderTest.java (hunk 3 above)
Comment (vibhorgarg) [reply tef-impl/src/test/java/flipkart/tef/execution/FluentCapabilityBuilderTest.java:198]: aded

File: tef-impl/src/test/java/flipkart/tef/execution/FluentCapabilityBuilderTest.java (hunk 2 above)
Comment (vibhorgarg) [reply tef-impl/src/test/java/flipkart/tef/execution/FluentCapabilityBuilderTest.java:215]: done

File: pom.xml (hunk 1 above)
Comment (vibhorgarg) [reply pom.xml]: Reducing to a minor version bumnp only, as we will have to make a separate pr (and maybe a separate pom) that compiles using java 17 to qualify for a mjor version bump

**Review by bageshwar — APPROVED** [review]

**Pull request merged**